app/models/concerns/course/assessment/submission/todo_concern.rb

```diff
@@ -22,9 +22,13 @@ def update_todo
     elsif submitted? || graded? || published?
       todo.update_column(:workflow_state, 'completed') unless todo.completed?
     end
+  rescue ActiveRecordError => error
```

Comment:
`update_column` will only raise in rare cases (see below). In those cases it might make sense for it to raise a rollback as well. 

> This method raises an ActiveRecord::ActiveRecordError when called on new objects, or when at least one of the attributes is marked as readonly.

[Link](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-update_columns)